date-mod.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+var myObj = (function iife(){
+  //sets the day of the month to the date object
+  function setDate(){
+    var sd = new Date();
+    
+    return sd.setDate(15);
+  }
+  //method returns the day of the week (from 0 to 6) for the specified date.
+  function getDayName(){
+    var d = new Date();
+    var weekday = new Array(7);
+    weekday[0]=  'Sunday';
+    weekday[1] = 'Monday';
+    weekday[2] = 'Tuesday';
+    weekday[3] = 'Wednesday';
+    weekday[4] = 'Thursday';
+    weekday[5] = 'Friday';
+    weekday[6] = 'Saturday';
+    var gda = d.getDay();
+    //return weekday[gda];
+    
+    return weekday[gda];
+  }
+  //returns the month (from 0 to 11) for the
+  //specified date, according to local time.
+  function getMonthName(){
+    var d = new Date();
+    var month = new Array(12);
+    month[0]=  'January';
+    month[1] = 'February';
+    month[2] = 'March';
+    month[3] = 'April';
+    month[4] = 'May';
+    month[5] = 'June';
+    month[6] = 'July';
+    month[7] = 'August';
+    month[8] = 'September';
+    month[9] = 'October';
+    month[10] = 'November';
+    month[11] = 'December';
+    var gm = d.getMonth();
+    
+    return month[gm];
+  }
+  //var d = new Date();
+  //var formatted = getMonthName + getDay + var yy = d.getFullYear();
+  ////returns the day of the month (from 1 to 31) for the specified date.
+  //function getDate({ format: "formatted" })
+  function getDate(){
```

Comment:
This function needs to take a parameter. There are three possibilities you want to handle: 1. the user doesn't supply and argument. In that scenario, return the millisecond representation of the date object you should be storing internally. 2. The user provides and object with a format property and the value of that property is "milliseconds" - this case you'll also return the date represented as milliseconds:

`dateObj.getDate({ format: "milliseconds" });`

And finally, an object with a property called format and a value for that property of "formatted":

`dateObj.getDate({ format: "formatted" });`

This version should return the date as a formatted string that looks like the following: "April 26, 2016"